source/extensions/filters/network/thrift_proxy/binary_protocol_impl.cc

```diff
@@ -265,6 +265,106 @@ bool BinaryProtocolImpl::readBinary(Buffer::Instance& buffer, std::string& value
   return readString(buffer, value);
 }
 
+void BinaryProtocolImpl::writeMessageBegin(Buffer::Instance& buffer, const std::string& name,
+                                           MessageType msg_type, int32_t seq_id) {
+  BufferHelper::writeU16(buffer, Magic);
+  BufferHelper::writeU16(buffer, static_cast<uint16_t>(msg_type));
+  writeString(buffer, name);
+  BufferHelper::writeI32(buffer, seq_id);
+}
+
+void BinaryProtocolImpl::writeMessageEnd(Buffer::Instance& buffer) {
+  UNREFERENCED_PARAMETER(buffer);
+}
+
+void BinaryProtocolImpl::writeStructBegin(Buffer::Instance& buffer, const std::string& name) {
+  UNREFERENCED_PARAMETER(buffer);
+  UNREFERENCED_PARAMETER(name);
+}
+
+void BinaryProtocolImpl::writeStructEnd(Buffer::Instance& buffer) {
+  UNREFERENCED_PARAMETER(buffer);
+}
+
+void BinaryProtocolImpl::writeFieldBegin(Buffer::Instance& buffer, const std::string& name,
+                                         FieldType field_type, int16_t field_id) {
+  UNREFERENCED_PARAMETER(name);
+
+  BufferHelper::writeI8(buffer, static_cast<uint8_t>(field_type));
+  if (field_type == FieldType::Stop) {
+    return;
+  }
+
+  BufferHelper::writeI16(buffer, field_id);
```

Comment:
Negative values for `field_id` are okay, right?

Reply:
According to the protocol specs, it's a signed, 16-bit integer. The code generator will not let you choose negative values or 0. So I think I'll allow negative values, just in case.